lib/fluent/plugin/in_s3.rb

```diff
@@ -1,115 +1,107 @@
 require 'fluent/input'
 
-module Fluent
-  class S3Input < Input
+require 'aws-sdk-resources'
+require 'zlib'
+require 'time'
+require 'tempfile'
+
+module Fluent::Plugin
+  class S3Input < Fluent::Plugin::Input
     Fluent::Plugin.register_input('s3', self)
 
+    helpers :compat_parameters, :parser, :thread
+
     def initialize
       super
-      require 'aws-sdk-resources'
-      require 'zlib'
-      require 'time'
-      require 'tempfile'
-
       @extractor = nil
     end
 
-    # For fluentd v0.12.16 or earlier
-    class << self
-      unless method_defined?(:desc)
-        def desc(description)
-        end
-      end
-    end
-    unless Fluent::Config::ConfigureProxy.method_defined?(:desc)
-      Fluent::Config::ConfigureProxy.class_eval do
-        def desc(description)
-        end
-      end
-    end
+    DEFAULT_PARSE_TYPE = "none"
```

Comment:
This should be `out_file`: https://github.com/fluent/fluent-plugin-s3/blob/4ed5d9f86ec0b07e41738e6e00398ad88a356b2a/lib/fluent/plugin/out_s3.rb#L97